lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -1448,25 +1448,45 @@ void ApplyWorkingDir(SmallString &clang_argument, StringRef cur_working_dir) {
   llvm::sys::path::append(clang_argument, cur_working_dir, rel_path);
   llvm::sys::path::remove_dots(clang_argument);
 }
+
+enum ClangArgCategory {
+  ClangArgCategoryNone = 0,
+  ClangArgCategoryUnique = 1 << 0,
+  ClangArgCategoryMacro = 1 << 1,
+};
+
+inline bool IsMultiArgClangFlag(const StringRef& arg) {
+  return arg == "-D" || arg == "-U" || arg == "-working-directory";
+}
+
+ClangArgCategory GetClangArgCategory(const StringRef& arg) {
```

Comment:
Why not split this into two separate functions that each return a bool?

Reply:
In addition to https://github.com/apple/llvm-project/pull/1444#discussion_r453773505 I can note that this way (with one function) we're checking `arg.size() <= 2 || arg[0] != '-'` condition once, not twice. Moreover, splitting this in two functions would case duplication of this condition or separating it with third function for which I'm struggling to propose a good name :)

Reply:
I would prefer just having

```
bool IsMacroDefinition(StringRef arg) { return arg.startswith("-D") || arg.startswith("-U"); }
bool shouldUnique(StringRef arg) { return IsMacroDefinition(arg) || arg.startswith("-working-directory"); }

```

It's shorter, easy to read, and likely results in very similar code after inlining.

Reply:
Ok, looks good to me either. I was just a little afraid that it was some kind of [important] optimization in old version of code, and tried to save it :)

Reply:
I should note that `-working-directory` is not something we want to unique as we don't pass this arg to clang at all but use it ourselves to make absolute paths